### PR TITLE
fix(test): fix the corner case for raft entries test

### PR DESCRIPTION
**Pull request by mangalaman93**
A failure here: https://github.com/dgraph-io/dgraph/actions/runs/3952517043/jobs/6767694064

This happens when the length of data byte slice is 0. I do not think that is a valid case.

```
=== RUN   TestEntryReadWrite
    encryption_test.go:48: 
        	Error Trace:	encryption_test.go:48
        	Error:      	Not equal: 
        	            	expected: []byte{}
        	            	actual  : []byte(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	-([]uint8) {
        	            	-}
        	            	+([]uint8) <nil>
        	            	 
        	Test:       	TestEntryReadWrite
--- FAIL: TestEntryReadWrite (0.02s)
FAIL
```

### CODE REVIEW

**Review by sanjayk-github-dev — DISMISSED** [review]
Looks good to me

**Review by anurags92 — APPROVED** [review]

**Review by sanjayk-github-dev — APPROVED** [review]
LGTM

**Review by anurags92 — APPROVED** [review]
